nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/dao/impl/StandardProcessGroupDAO.java

```diff
@@ -124,19 +126,30 @@ public boolean hasProcessGroup(String groupId) {
 
     @Override
     public void verifyUpdate(final ProcessGroupDTO processGroup) {
+        final ProcessGroup group = locateProcessGroup(flowController, processGroup.getId());
+        verifyCanSetParameterContext(processGroup, group);
+
+        final String executionEngine = processGroup.getExecutionEngine();
+        if (executionEngine != null) {
+            group.verifyCanSetExecutionEngine(ExecutionEngine.valueOf(executionEngine));
+        }
+
+        final VersionControlInformationDTO versionControlInfoDTO = processGroup.getVersionControlInformation();
+        final VersionControlInformation versionControlInformation = group.getVersionControlInformation();
+        if (versionControlInfoDTO != null && versionControlInformation != null) {
+            throw new IllegalStateException("Cannot set Version Control Info because process group is already under version control");
```

Comment:
Do we also want throw an exception if the process group has a child process group with version control information? I know we currently allow nested versioning but I feel like we may want to prevent it in this case.

Reply:
Updated